src/mono/CMakeLists.txt

```diff
@@ -510,7 +510,7 @@ endif()
 
 # Decide if we need zlib-ng.
 if(NOT DISABLE_EMBEDDED_PDB OR NOT DISABLE_LOG_PROFILER_GZ)
-  if (CLR_CMAKE_USE_SYSTEM_ZLIB)
+  if (CLR_CMAKE_USE_SYSTEM_ZLIB OR CLR_CMAKE_HOST_ARCH_WASM)
     # if we're not on a platform where we use the in-tree zlib-ng, require system zlib
     include(${CLR_SRC_NATIVE_DIR}/libs/System.IO.Compression.Native/extra_libs.cmake)
     set(Z_LIBS)
```

Comment:
Here is where the value of `Z_LIBS` is set for all the CMakeLists.txt in the subfolders to use.